libs/common/src/platform/misc/utils.ts

```diff
@@ -538,6 +538,17 @@ export class Utils {
     return of(undefined).pipe(switchMap(() => generator()));
   }
 
+  /**
+   * Return the number of days remaining before a target date arrives.
+   * Returns 0 if the day has already passed.
+   */
+  static daysRemaining(targetDate: Date): number {
+    const diffTime = targetDate.getTime() - Date.now();
+    const [ms, seconds, minutes, hours] = [1000, 60, 60, 24];
+    const msPerDay = ms * seconds * minutes * hours;
```

Comment:
Why allocate and calculate this every time the method is called? It's a constant (86400000)
```suggestion
    const msPerDay = 86400000;
```